sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasing.scala

```diff
@@ -39,6 +39,14 @@ object NestedColumnAliasing {
           NestedColumnAliasing.replaceToAliases(plan, nestedFieldToAlias, attrToAliases)
       }
 
+    case Project(projectList, Filter(condition, child))
```

Comment:
I think we better leave a few comment explaining this case.

Reply:
+1